docs/contributors/folder-structure.md

```diff
@@ -0,0 +1,192 @@
+# Folder Structure
+
+The following snippet explains how the WooCommerce Blocks repository is structured omitting irrelevant or obvious items with further explanations:
+
+    │
+    ├── LICENSE
+    ├── README.md
+    ├── .github/CODEOWNERS
+    ├── .github/CODE_OF_CONDUCT.md
+    ├── .github/CONTRIBUTING.md
+    ├── .github/SECURITY.md
+    │
+    ├── .distignore
+    ├── .editorconfig
+    ├── .eslintignore
+    ├── .eslintrc
+    ├── .gitattributes
+    ├── .gitignore
+    ├── .prettierrc.js
+    ├── .stylelintrc.json
+    ├── phpcs.xml.dist
+    │   Dot files and config files used to configure the various linting tools
+    │   used in the repository (PHP, JS, styles...).
+    │
+    ├── .nvmrc
+    │   Required Node version of the project.
+    │
+    ├── babel.config.js
+    ├── global.d.ts
+    ├── tsconfig.base.json
+    ├── tsconfig.json
+    ├── webpack.config.js
+    │   Transpilation and bundling config files.
+    │
+    ├── phpcs.xml
+    │   Config file for the PHP Coding Standards.
+    │
+    ├── phpunit.xml.dist
+    │   Config file for PHPUnit.
+    │
+    ├── postcss.config.js
+    │   Config file for PostCSS.
+    │
+    ├── renovate.json
+    │   Config file for Renovate.
+    │
+    ├── docker-compose.yml
+    │   Docker config files for the development and testing environment.
+    │
+    ├── .env
+    ├── .wp-env.json
+    │   Config files for the development and testing environment.
+    │   Includes WordPress, the WooCommerce plugin and the Storefront theme.
+    │
+    ├── composer.lock
+    ├── composer.json
+    │   Handling of PHP dependencies. Essentially used for development tools.
+    │   The production code don't use external PHP dependencies.
+    │
+    ├── package-lock.json
+    ├── package.json
+    │	Handling of JavaScript dependencies. Both for development tools and
+    │   production dependencies. The package.json also serves to define common
+    |   tasks and scripts used for day to day development.
+    │
+    ├── readme.txt
+    │   Readme of the WooCommerce Blocks plugin hosted on the WordPress
+    │   plugin repository.
+    │
+    ├── woocommerce-gutenberg-products-block.php
+    │   Entry point of the WooCommerce Blocks plugin.
+    │
+    ├── .github/*
+    │   Config of the different Github features (issues and PR templates, CI, 
+    │   owners).
+    │
+    ├── .sources/*
+    │   Sketch files of the WooCommerce Blocks plugin.
+    │
+    ├── .wordpress-org/*
+    │   Assets of the WooCommerce Blocks plugin hosted on the WordPress plugin
+    │   repository.
+    │
+    ├── assets/css/*
+    │   The SCSS files of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/*
+    │   The React components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/atomic
+    │   The atomic components such as product title, product rating, product
+    │   image, etc. These atomic components are used by various blocks.
```

Comment:
Used by the product blocks.

Reply:
Thanks for clarifying this, @mikejolley!